modules/apps/transfer/keeper/relay.go

```diff
@@ -277,6 +278,10 @@ func (k Keeper) OnRecvPacket(ctx sdk.Context, packet channeltypes.Packet, data t
 		return errorsmod.Wrap(err, "failed to mint IBC tokens")
 	}
 
+	if !k.bankKeeper.HasDenomMetaData(ctx, voucherDenom) {
+		k.SetDenomMetadata(ctx, denomTrace)
```

Comment:
tests should be modified to assert metadata is set

Reply:
Added the assertions.